app/controllers/des/DesController.scala

```diff
@@ -37,7 +38,16 @@ class DesController @Inject() (
   def getFinancialData(vrn: Vrn): Action[AnyContent] = Action.async { implicit request =>
     for {
       fd <- desConnector.getFinancialData(vrn)
-    } yield (Ok(Json.toJson(fd)))
+    } yield {
+
+      val filtered: Seq[Transaction] = fd.financialTransactions.filter(f => (f.chargeType == "VAT Return Credit Charge" || f.chargeType == "VAT Return Debit Charge"))
```

Comment:
could you extranct the predicate so it's more clear what are we filtering? So the could would look similar to this:

```
def isVatSomething(chargeType: ChargeType) : Boolean  = f.chargeType == "VAT Return Credit Charge" || f.chargeType == "VAT Return Debit Charge"

val onlyVatSomething = fd.financialTransactions.filter(isVatSomething)
```